src/evict/evict_page.c

```diff
@@ -661,9 +664,24 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
+    saved_pinned_id = WT_SESSION_TXN_SHARED(session)->pinned_id;
+
+    /* Acquire a snapshot if coming through eviction thread route. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL)) {
+        __wt_txn_get_snapshot(session);
+        release_snapshot = true;
+    }
```

Comment:
Do we need to change the isolation level of application threads to read committed when it is doing eviction?